pkgs/applications/networking/cluster/k3s/README.md

```diff
@@ -71,3 +71,44 @@ In order to resolve this issue, we propose backporting not just new patch releas
 
 In the above example, where NixOS 23.05 included k3s 1.26, and 23.11 included k3s 1.28, that means we would backport 1.27 to the NixOS 23.05 release, and backport all patches for 1.26 and 1.27.
 This would allow someone to upgrade between those NixOS releases in a supported configuration.
+
+
+## K3s upkeep for nixpkgs maintainers
+
+* Whenever adding a new major/minor K3s version to nixpkgs:
+  - update `k3s` alias to the latest version.
+  - add a NixOS release note scheduling the removal of all K3s packages but the latest
+  - include migration information from both Kubernetes and K3s projects
+
+* After every NixOS release, all K3s versions but the latest should be removed by K3s maintainers.
```

Comment:
I'd clarify this point a little bit more, to avoid having the problem we had in the past (remember https://github.com/NixOS/nixpkgs/issues/213943#issue-1565759962?):

```suggestion
* After every NixOS release, all K3s versions but the latest should be removed by K3s maintainers.
  This way, next NixOS release will have at least one K3s release in common with the previous
  NixOS release. This will provide a smooth upgrade path for users.
```

Reply:
Before a release there might exist multiple K3s versions.
After a release there is going to exist only a single K3s version.
While I understand what you are saying, that is not to be part of 'after a release'.
After a release => Unstable
Unstable starts with a single release of K3s.
Stable will have multiple K3s versions to provide the smooth upgrade path.

Reply:
@yajo Thanks for your suggestion.

I have tried to integrate what you said. Let me know how it went.